app.rb

```diff
@@ -0,0 +1,128 @@
+  end
+
+  def list_books
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'There are no people in the library'
+    else
+      @people.each do |person|
+        if person.instance_of?(Student)
+          puts "[Student]: Name: #{person.name}, Id: #{person.id}, Age: #{person.age}"
+        else
+          puts "[Teacher]: Name: #{person.name}, Id: #{person.id}, Age: #{person.age}"
+        end
+      end
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = gets.chomp
+
+    case person_type
+    when '1'
+      student = create_student
+      @people << student
+      puts 'Person created successfully'
+
+    when '2'
+      teacher = create_teacher
+      @people << teacher
+      puts 'Person created successfully'
+    else
+      puts 'Invalid option'
+    end
+  end
+
+  def create_student
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Classroom:'
+    classroom = Classroom.new(gets.chomp)
+
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp
+    if parent_permission == 'Y' || parent_permission.downcase == 'y'
+      parent_permission = true
+    elsif parent_permission == 'N' || parent_permission.downcase == 'n'
+      parent_permission = false
+    else
+      return nil
+    end
+
+    Student.new(age, classroom, parent_permission, name)
+  end
+
+  def create_teacher
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    Teacher.new(age, specialization, name)
+    puts 'Teacher created successfullly'
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    @books << Book.new(title, author)
+    puts 'Book created successfully'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+
+    book_index = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age} "
+    end
+
+    person_index = gets.chomp.to_i
+
+    print 'Date: '
+    date = gets.chomp
+
+    @rentals << Rental.new(date, @books[book_index], @people[person_index])
+    puts 'Rental created successfully'
+  end
+
+  def list_rentals_by_person_id
+    print 'ID of person: '
+    id = gets.chomp.to_i
+
+    rentals = @rentals.filter { |rental| rental.person.id == id }
+    puts 'Rentals:'
+    rentals.each { |rental| puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}" }
+  end
```

Comment:
- [ ] Without being able to create a `Person` it is not possible to verify the methods `def create_rental`, and ` def list_rentals_by_person_id` since they are connected.

![image](https://github.com/danielmatama/OOP-library/assets/78251927/d0debd83-53c5-4f0d-b95b-4e8fd1270368)

- [ ] Please after fixing the issues mentioned concerning creating a Person for both the `Student` and `Teacher`. kindly ensure that it is also possible to create a rental and that the console app works as expected. 👍🤗